license-maven-plugin-svn/src/main/java/com/mycila/maven/plugin/license/svn/SVNPropertiesProvider.java

```diff
@@ -41,46 +43,60 @@
  * @author Matthieu Brouillard
  */
 public class SVNPropertiesProvider implements PropertiesProvider {
-  ThreadLocal<SVNClientManager> svnClientThreadLocal = new ThreadLocal<>();
-  ThreadLocal<SimpleDateFormat> sdfTimestampThreadLocal = new ThreadLocal<SimpleDateFormat>() {
-    @Override
-    protected SimpleDateFormat initialValue() {
-      return new SimpleDateFormat("yyyyMMdd-HH:mm:ss");
-    }
-  };
+
 
   public static final String SVN_COPYRIGHT_LASTCHANGE_YEAR_KEY = "license.svn.lastchange.year";
   public static final String SVN_COPYRIGHT_LASTCHANGE_DATE_KEY = "license.svn.lastchange.date";
   public static final String SVN_COPYRIGHT_LASTCHANGE_TIMESTAMP_KEY = "license.svn.lastchange.timestamp";
   public static final String SVN_COPYRIGHT_LASTCHANGE_REVISION_KEY = "license.svn.lastchange.revision";
   public static final String SVN_COPYRIGHT_YEARS_RANGE_KEY = "license.svn.years.range";
-
   public static final String INCEPTION_YEAR_KEY = "project.inceptionYear";
-
   public static final String SVN_SERVER_ID_PLUGIN_KEY = "license.svn.serverId";
 
-  private volatile boolean initialized = false;
+  private Credentials svnCredentials;
+
+  private final AtomicBoolean warnedIfShallow = new AtomicBoolean();
+  private final Queue<SVNClientManager> clients = new ConcurrentLinkedQueue<>();
+  private final ThreadLocal<SimpleDateFormat> sdfTimestampThreadLocal = ThreadLocal.withInitial(
+      () -> new SimpleDateFormat("yyyyMMdd-HH:mm:ss"));
+  private final ThreadLocal<SVNClientManager> svnClientThreadLocal = ThreadLocal.withInitial(() -> {
+    SVNClientManager svnClientManager = svnCredentials == null ?
+        SVNClientManager.newInstance(new DefaultSVNOptions()) :
+        SVNClientManager.newInstance(new DefaultSVNOptions(), svnCredentials.getLogin(),
+            svnCredentials.getPassword());
+    clients.offer(svnClientManager);
+    return svnClientManager;
+  });
```

Comment:
Previously incorrect resource cleanup: thread-local SVN clients were not closed. Kee ing track of all created clients per thread to clean them at the end.